Room/app/src/main/java/com/example/room/presentation/main/MainActivity.kt

```diff
@@ -0,0 +1,58 @@
+package com.example.room.presentation.main
+
+import androidx.appcompat.app.AppCompatActivity
+import android.os.Bundle
+import android.widget.Toast
+import androidx.activity.viewModels
+import androidx.databinding.DataBindingUtil
+import androidx.lifecycle.Observer
+import androidx.recyclerview.widget.LinearLayoutManager
+import com.example.room.R
+import com.example.room.databinding.ActivityMainBinding
+import com.example.room.data.local.db.Friend
+import com.example.room.presentation.main.adapter.FriendRecycleViewAdapter
+import com.example.room.presentation.main.viewmodel.FriendViewModel
+import dagger.hilt.android.AndroidEntryPoint
+
+@AndroidEntryPoint
+class MainActivity : AppCompatActivity() {
+
+    private lateinit var binding: ActivityMainBinding
+    private val friendViewModel by viewModels<FriendViewModel>()
+    private lateinit var friendAdapter: FriendRecycleViewAdapter
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        binding = DataBindingUtil.setContentView(this, R.layout.activity_main)
+
+        binding.friendViewModel = friendViewModel
+        binding.lifecycleOwner = this
+        initRecyclerView()
+
+        friendViewModel.message.observe(this, Observer { event ->
+            event.getContentIfNotHandled()?.let {
+                Toast.makeText(this, it, Toast.LENGTH_SHORT).show()
+            }
+        })
+    }
+
+    private fun initRecyclerView() {
+        binding.friendRecyclerView.layoutManager = LinearLayoutManager(this)
+        friendAdapter = FriendRecycleViewAdapter { selectedItem: Friend ->
+            listItemClicked(selectedItem)
+        }
```

Comment:
```suggestion
        friendAdapter = FriendRecycleViewAdapter(::listItemClicked)
```
오호라,, 고차 함수까지 접수하셨군요! 대단하십니다 👏👏
위와 같이 표현이 가능하다는 것도 알아두시면 좋을 것 같아요오

Reply:
더블콜론 명시하겠씀다!!